### PR TITLE
Odds

### PR DESCRIPTION
**Pull request by HEnquist**
Add support for odd lengths.
Also a big rewrite to be more similar to RustFFT, with a Planner etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,94 +1,120 @@
...
+
+// make a planner
+let mut real_planner = RealFftPlanner::<f64>::new();
 
-//create an FFT and forward transform the input data
-let mut r2c = RealToComplex::<f64>::new(256).unwrap();
+// create an FFT
+let r2c = real_planner.plan_fft_forward(length);
+// make input and output vectors
+let mut indata = r2c.make_input_vec();
```
Comment (ejmahler) [inline README.md:98]: this is a neat idea

File: src/lib.rs [hunk 2]
```diff
@@ -189,343 +172,333 @@ where
...
-                        length,
-                        fft,
-                        buffer_out,
-                        scratch,
-                        is_even: true,
-                    })
-                }
-            }
+impl<T: FftNum>  RealToComplex<T> {
+    /// Create a new RealToComplex FFT for input data of a given length. Returns an error if the length is not even.
```
Comment (ejmahler) [inline src/lib.rs]: This comment isn't accurate anymore, since it always returns Ok()

Comment (HEnquist) [reply src/lib.rs]: That looks as it's not from the latest version, this part changed a lot in the next couple of commits.

File: README.md (hunk 1 above)
Comment (HEnquist) [reply README.md:98]: Yes I think this is quite convenient, so avoid writing the same "let blahblah = vec![.....]" over and over. Something you would like to have in RustFFT as well?

File: src/lib.rs (hunk 2 above)
Comment (HEnquist) [reply src/lib.rs]: And thanks for looking!

**Pull request merged**